napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+- `[in]  data`: Optional arbitrary context data to be passed in to the callback when the function is invoked
+- `[out] result`: A `napi_value` representing a JavaScript Function
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns a N-API value corresponding to a JavaScript Function object.
+It's used to wrap native functions so that they can be invoked from JavaScript.
+
+JavaScript Functions are described in [Section 19.2](https://tc39.github.io/ecma262/#sec-function-objects) 
+of the ECMAScript Language Specification.
+
+### *napi_create_number*
+
+#### Signature
+```
+napi_status napi_create_number(napi_env env, double value, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: Double-precision value to be represented in JavaScript
+- `[out] result`: A `napi_value` representing a JavaScript Number
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to convert from C++ doubles to JavaScript Number.
+
+JavaScript Number type is described in [Section 6.1.6](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-number-type) 
+of the ECMAScript Language Specification.
+
+### *napi_create_object*
+
+#### Signature
+```
+napi_status napi_create_object(napi_env env, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[out] result`: A `napi_value` representing a JavaScript Object
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API allocates a default JavaScript Object. 
+It is the equivalent of doing `new Object()` in JavaScript.
+
+JavaScript Object type is described in [Section 6.1.7](https://tc39.github.io/ecma262/#sec-object-type) of the ECMAScript Language Specification.
+
+### *napi_create_string_utf16*
+
+#### Signature
+```
+napi_status napi_create_string_utf16(napi_env env, const char16_t* str, size_t length, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  str`: Character buffer representing a UTF16-LE encoded string
+- `[in]  length`: The length of the string in characters, or -1 if it is null-terminated.
+- `[out] result`: A `napi_value` representing a JavaScript String
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API creates a JavaScript String object from a UTF16-LE encoded C++ string
+
+JavaScript String type is described in [Section 6.1.4](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-string-type) of the ECMAScript Language Specification.
+
+### *napi_create_string_utf8*
+
+#### Signature
+```
+napi_status napi_create_string_utf8(napi_env env, const char* str, size_t length, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  s`: Character buffer representing a utf8 encoded string
```

Comment:
need to join "encoded" to the previous word with a hyphen here. We need to agree on the capitalization of "utf8" and "utf16-le".